### ISSUE TITLE
Aconcagua (Japan) (Disc 1 & 2)

### ISSUE THREAD

**Issue by AmuroRayMan** [socram8888/tonyhax#40]
**tonyhax version**: v1.2.2b (think it was the build posted with the Parasite Eve fix)

**Game name**: Aconcagua (Japan) (Disc 1 & 2)

**Game code**: SCPS-10131 / SCPS-10132

**Console model**: SCPH-101 NTSC U/C, BIOS v4.5 05/25/00 A

**Executable hash**: 
Disc 1
SCPS_101.31 SHA-1: **0AA2AF9D40270955DF9BBFBC003402DF7B940956**
PROGRAM.BIN SHA-1: **FD51721B8433F33A15135EBE2342713DA53DBA52**

Disc 2
SCPS_101.32 SHA-1: **0AA2AF9D40270955DF9BBFBC003402DF7B940956**
PROGRAM.BIN SHA-1: **FD51721B8433F33A15135EBE2342713DA53DBA52**

After some digging there's a PS-X EXE header for PROGRAM.BIN and that's where the antipiracy message is.

**Triggered at**: Immediately
I don't have a copy of disc 2 ready to go at the moment, but disc 1 triggers it pretty much immediately after the game starts loading.

**Comment by socram8888** [comment]
Thanks for taking the time to properly fill the form! Will have a look and report back.

**Comment by socram8888** [comment]
This was another though one: https://github.com/socram8888/tonyhax/blob/master/patcher.c#L27

Please try with this beta version: [tonyhax-v1.2.3b.zip](https://github.com/socram8888/tonyhax/files/6237733/tonyhax-v1.2.3b.zip) (compiled from 485246775a7b38bbf5be23a4dce604c9bf9df7fa)

**Comment by socram8888** [comment]
I'll take that thumbs as "working".

**Comment by AmuroRayMan** [comment]
> 
> 
> I'll take that thumbs as "working".

Oh sorry my bad, I have this habit of marking things like that to let them know I saw it. I tested it and it loads the patch but it still triggers the protection.

**Comment by socram8888** [comment]
So it says on the screen that it is "Patching" it?

**Comment by AmuroRayMan** [comment]
Indeed, it finds the checksum in the list of patches, patches it, executes the game and then still immediately goes to the antipiracy message.

**Comment by socram8888** [comment]
I found out why - the game executes the antipiracy check multiple times in multiple executables. Ugh.

**Comment by socram8888** [comment]
Still, wait - do you mean you can't even get to the title screen? That shouldn't be possible.

**Comment by AmuroRayMan** [comment]
Yeah it immediately triggers it, as soon as the game executes it shows the antipiracy screen.

**Comment by AmuroRayMan** [comment]
So based on a previous report I decided to look on google to see if there were any gameshark codes, and found one that claims to let it work on modchip consoles:
`D0027756 1040
80027756 1000`
Not sure if it'll help, but thought I'd post it as well.

**Comment by socram8888** [comment]
Okay, I've found a way to tweak no$psx to emulate having a modchip:
![imagen](https://user-images.githubusercontent.com/515068/113275665-88255b80-92ce-11eb-8051-bf61bd1614cd.png)
Still, when launching via tonyhax it works just fine:
![imagen](https://user-images.githubusercontent.com/515068/113275739-9f644900-92ce-11eb-89a4-e459421e0017.png)
![imagen](https://user-images.githubusercontent.com/515068/113275750-a3906680-92ce-11eb-95c2-5bf6e3c05721.png)

**Comment by AmuroRayMan** [comment]
Okay so I'm noticing the boot process doesn't have the 
`Load: 80010008`
`Load: 00014000`
part after Loading executable for me. 
![tonyhax_bug](https://user-images.githubusercontent.com/1740249/113335456-cffdb000-92d9-11eb-830b-f63cac809617.png)
It also goes to an English message when it triggers the protection, is that due to my console's region?

**Comment by socram8888** [comment]
Yeah the "Load message" was from a test I had made, where I was getting inconsistent CRCs due to some no$psx quirks. I tried recompiling again without that message and it still worked fine, though.

The reason I am getting a Japanese error is because my console region is being detected as an unknown and falling back to Japanese, as I am using the no$psx BIOS which has a space where the American one has a "U".

I will try using an actual console and see how it goes for me.

**Comment by socram8888** [comment]
I've still haven't had time to get around testing on real hardware, but could you please try with this beta version?

[tonyhax-v1.3b.zip](https://github.com/socram8888/tonyhax/files/6291888/tonyhax-v1.3b.zip) (compiled from 1e587653ae044c477a6d76a5d73c0c896dc05554)

I think the reason it failed to work but did on software was a cache coherence issue. This _might_ fix it.

**Comment by AmuroRayMan** [comment]
Gotta dig a PS2 out to copy over the new version, but I'll update when I'm able to test. Thank you for continuing to look into the problem.

**Comment by AmuroRayMan** [comment]
Still not working and there's a new issue with that build. When it attempts to execute tonyhax, it makes my video output stop working for a bit (never happened on older versions), but the program is still running. And I decided to just put the disc in and it booted (video came back after this) but still triggers the protection. Perhaps there was an issue and it attempted to switch to PAL video instead of NTSC?

**Comment by socram8888** [comment]
Is your console chipped? I tried yesterday this very game on my SCPH-102 and it booted just fine.

Does the protection trigger immediately or upon loading a particular savefile or something like that? This game seems to have the antipiracy in multiple places and I am only patching the very first one, for testing basically.

The issue with the video is known and I've already fixed it in current master.
